scripts/globals/chocobo_digging.lua

```diff
@@ -801,209 +828,344 @@ local digInfo =
-    sum = 0
+    local pick   = math.random(sum)
+    sum          = 0
 
     for i = 1, #possibleItems do
-        sum = sum + possibleItems[i][2]
+        local itemWeight = possibleItems[i][2]
+
+        -- Skill-up weight variation and ore moon variation.
+        if DigRank > 0 then
+            if itemWeight >= 150 then
+                itemWeight = itemWeight * (0.95^DigRank) 
+            elseif itemWeight >= 125 then
+                itemWeight = itemWeight * (0.97^DigRank)
+            elseif itemWeight >= 100 then     
+                itemWeight = itemWeight * (0.99^DigRank)
+            elseif itemWeight >= 50 then
+                itemWeight = itemWeight * (1.03^DigRank)
+            elseif itemWeight >= 1 then
+                itemWeight = itemWeight * (1.05^DigRank)
+            end
+        end
+
+        if possibleItems[i][1] == 1255 then
+            if moon >= 7 and moon <= 9 then
+                itemWeight = itemWeight * 0.7
+            elseif moon >= 10 and moon <= 14 then
+                itemWeight = itemWeight
+            elseif moon >= 15 and moon <= 21 then
+                itemWeight = itemWeight * 0.9
+            elseif moon >= 21 and moon <= 24 then
+                itemWeight = itemWeight * 0.5
+            end
+        end
+
+        sum = sum + itemWeight
+
         if sum >= pick then
             itemId = possibleItems[i][1]
             break
         end
     end
 
-    -- item is a crystal or ore
+    -- Item is a crystal or ore
     if itemId == 4096 then
-        if weather >= xi.weather.HOT_SPELL then
-            itemId = crystalMap[weather]
-        else
-            itemId = 0
-        end
+        itemId = crystalMap[weather]
     elseif itemId == 1255 then
-        if weather >= xi.weather.CLOUDS and moon >= 10 and moon <= 40 and player:getSkillRank(xi.skill.DIG) >= 7 then
-            itemId = oreMap[VanadielDayOfTheWeek()]
-        else
-            itemId = 0
-        end
+        itemId = oreMap[VanadielDayOfTheWeek()]
     end
 
     return itemId
 end
 
+-- Main function.
 xi.chocoboDig.start = function(player, precheck)
+    -- Make sure the player can dig before going any further.
+    if not canDig(player) then
+        return
+    end
+
+    -- Handle AMK07 first thing.
     local zoneId = player:getZoneID()
-    local text = zones[zoneId].text
+    local text   = zones[zoneId].text
 
-    -- make sure the player can dig before going any further
-    -- (and also cause i need a return before core can go any further with this)
-    if precheck then
-        return canDig(player)
+    if
+        xi.settings.ENABLE_AMK == 1 and
+        player:getCurrentMission(xi.mission.log_id.AMK) == xi.mission.id.amk.SHOCK_ARRANT_ABUSE_OF_AUTHORITY and
+        xi.amk.helpers.chocoboDig(player, zoneId, text)
+    then
+        return
+    end
 
-    else
-        local roll = math.random(0, 100)
-        local moon = VanadielMoonPhase()
+    -- Handle distance automatic fail.
```

Comment:
my dig results from yesterday disagree with this. I was in same spot for several min, got multiple junk items. I didn't move.

Reply:
Correct.

This is custom behavior intended to reduce boting.
I will deal with it after we decide if this is something the base code could benefit from.

Reply:
sounds like module territory then

Reply:
Agree with teo, it's not our place to improve on retail in the base branch, that belongs in modules

Reply:
If thats the consensus, it shall be removed^^

Reply:
Can add it to an ASB module?